core/trino-main/src/test/java/io/trino/sql/planner/planprinter/TestExpressionAnonymization.java

```diff
@@ -0,0 +1,111 @@
+import io.trino.sql.tree.NullLiteral;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.sql.tree.SymbolReference;
+import io.trino.sql.tree.TimeLiteral;
+import io.trino.sql.tree.TimestampLiteral;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.sql.tree.ComparisonExpression.Operator.EQUAL;
+import static io.trino.sql.tree.ComparisonExpression.Operator.GREATER_THAN;
+import static io.trino.sql.tree.ComparisonExpression.Operator.LESS_THAN;
+import static io.trino.sql.tree.LogicalExpression.Operator.AND;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestExpressionAnonymization
+{
+    @Test
+    public void testSymbolReferenceAnonymization()
+    {
+        LogicalExpression expression = new LogicalExpression(AND, ImmutableList.of(
+                new ComparisonExpression(GREATER_THAN, new SymbolReference("a"), new LongLiteral("1")),
+                new ComparisonExpression(LESS_THAN, new SymbolReference("b"), new LongLiteral("2")),
+                new ComparisonExpression(EQUAL, new SymbolReference("c"), new LongLiteral("3"))));
+        HashingAnonymizer anonymizer = new HashingAnonymizer("test_environment");
+        assertThat(anonymizer.anonymize(expression))
+                .isEqualTo("((\"symbol_8580e4f2b35a15b4ebf8aa4f811f1088\" > 1) AND (\"symbol_eeb8ac2f5312455a9c9e764414238adf\" < 1) AND (\"symbol_cdc2b5337778e3defcfb85ad2dddfdf1\" = 1))");
+    }
+
+    @Test(dataProvider = "literals")
+    public void testLiteralAnonymization(Literal actual, Literal expected)
+    {
+        HashingAnonymizer anonymizer = new HashingAnonymizer("test_environment");
+        assertThat(anonymizer.anonymize(actual)).isEqualTo(expected.toString());
+    }
+
+    @DataProvider
+    public static Object[][] literals()
+    {
+        return new Object[][] {
+                {new BinaryLiteral("DEF321"), new BinaryLiteral("ABC123")},
+                {new StringLiteral("abc"), new StringLiteral("literal_f3e2a5a815a19f19e4dd00683d32bc64")},
+                {new GenericLiteral("bigint", "1"), new GenericLiteral("bigint", "literal_5be854d593624a18f2c4cf6a1f08ba9c")},
+                {new CharLiteral("a"), new CharLiteral("literal_8580e4f2b35a15b4ebf8aa4f811f1088")},
+                {new DecimalLiteral("123"), new DecimalLiteral("literal_65510eeb7822ced785732c8534f64e47")},
+                {new DoubleLiteral(String.valueOf(6554)), new DoubleLiteral("10000")},
+                {new DoubleLiteral(String.valueOf(Double.MAX_VALUE)), new DoubleLiteral("1E308")},
+                {new LongLiteral(String.valueOf(6554)), new LongLiteral("10000")},
+                {new LongLiteral(String.valueOf(Long.MAX_VALUE)), new LongLiteral("9223372036854775807")},
+                {new BooleanLiteral("true"), new BooleanLiteral("true")},
+                {new TimeLiteral("03:04:05"), new TimeLiteral("literal_777152a1ec1026feb3724588d1664773")},
```

Comment:
This is not a valid time literal. While it works because of lack of validation in the TimeLiteral class, that might not be the case in the future.